openapi/src/main/scala/endpoints/openapi/DocumentedBasicAuthentication.scala

```diff
@@ -0,0 +1,20 @@
+package endpoints
+package openapi
+
+import endpoints.algebra.BasicAuthentication.Credentials
+
+/**
+  * Interpreter for [[algebra.BasicAuthentication]] that produces
+  * OpenAPI documentation.
+  */
+trait DocumentedBasicAuthentication
+  extends algebra.DocumentedBasicAuthentication
+    with DocumentedEndpoints {
+
+  private[endpoints] def basicAuthentication: RequestHeaders[Credentials] =
```

Comment:
I just copied what is in the `endpoints.algebra.BasicAuthentication` trait. I don’t think it is new.

The problem is that because both interpreters and users of an algebra interface _extend_ this interface, using `protected` members is not useful to hide them from users. Hence the `private[endpoints]` qualifier.